### ISSUE TITLE
View bitmap not getting updated properly

### ISSUE THREAD

**Issue by SlidyBat** [ultralight-ux/ultralight#79]
**Summarize the Issue**
It seems like the dirty flag is being set on the view bitmap before the bitmap has actually been updated. This seems to only be an issue since c8d0fac; binaries before this commit seem to work fine

**Steps to Reproduce**
Use default driver and the following code:
```cpp
void Overlay::Paint()
{
    bool dirty = view()->is_bitmap_dirty();
    const RefPtr<Bitmap> bitmap = view()->bitmap();
    if (dirty)
    {
        CopyTexture(bitmap);
    }
    DrawTexture();
}

// ...

g_pOverlay = new Overlay();
g_pOverlay->view()->LoadHTML("<p id='my-text'>Hello</p>");

AddCommand("ul_test_js", []( const CommandArgs_t& args )
{
	g_pOverlay->view()->EvaluateScript( "document.getElementById('my-text').innerHTML = '" + args[1] + "'" );
} );
```
I've replicated the issue on this repo (use tilde key `~` to open console and run `ul_test_js <text>` command):
https://github.com/SlidyBat/BatEngine/tree/40256241d01e2ed4cd3adcc31ec128c48517d812

**Platform Info:**
 - OS: Windows (untested on other platforms)
 - SDK Version: Trunk

**Comment by SlidyBat** [comment]
Seems like my description of this issue is wrong, even if I update my texture from the bitmap every frame I still get the same issue. So the issue isn't that the dirty flag isn't getting set too early, the bitmap isn't getting updated at all until the next time the dirty flag is set.

Here's a visualization of what this looks like:
https://streamable.com/vtcm9

Every time I enter the command the bitmap should be updated with the text I feed it, but instead it gets updated to the text that it should have been updated to from the previous command.

**Comment by adamjs** [comment]
This is fixed in 1.0 (default GPUDriver now renders synchronously-- was previously using an asynchronous double-buffered approach that lagged a frame behind).
